### PR TITLE
Drop support for EOL Python 3.5

### PR DESCRIPTION
**Pull request by hugovk**
Python 3.5 reaches EOL on **2020-09-13**:

* https://devguide.python.org/#status-of-python-branches

The next Pillow release 8.0.0 release is due out after that, on **2020-10-15**.

This removes testing for Python 3.5 on the CIs, updates docs and metadata. I'll send another PR later with syntax updates, but first it's good to reduce the CI usage to speed up PR builds.

Edit: It also removes some redundant PyPy compatibility code for old and unsupported PyPy versions.

---

Notes:

* AppVeyor tests on two Python versions, the lowest supported (was 3.5, now 3.6) and the highest.

* Travis CI and GitHub Actions tested `PYTHONOPTIMIZE=2` on the 3.5 job (and `PYTHONOPTIMIZE=1` on 3.6). Now it tests `PYTHONOPTIMIZE=2` on the 3.7 job.

* Travis CI tested `s390x` on 3.5 (chosen arbitrarily). Now tests `s390x` on 3.8, as `arm64` is in 3.6 and `ppc64le` is on 3.7.

### CODE REVIEW

**Comment by hugovk** [comment]
Two of the Docker images are using Python 3.5:

* Debian 9 Stretch ([example](https://travis-ci.org/github/python-pillow/docker-images/jobs/701583804#L3679))
* Ubuntu 16.04 Xenial ([example](https://travis-ci.org/github/python-pillow/docker-images/jobs/701583801#L3264))

I think we can can probably drop the Docker image for Xenial, we're test Xenial extensively on Travis CI. (And Xenial's EOL is in April 2021, not _too_ far off.)

As for Stretch (EOL 2022), should we upgrade Python in the image? Have we done that before, or is the idea that the Docker images test the native Python only, and we can drop it?

**Comment by nulano** [comment]
There are also comments in winbuild readmes `winbuild/readme.md`:

> Python 3.6+ is required to generate valid scripts, but builds targeting Python 3.5+ are supported.

and docs `winbuild/docs.rst`:

> While the scripts can target any version of Python supported by Pillow, Python 3.6+ is required to generate valid build scripts.

**Comment by hugovk** [comment]
Thanks, updated.

**Comment by nulano** [comment]
> Not relevant for this particular PR, but two of the Docker images are using Python 3.5:
> 
> * Debian 9 Stretch ([example](https://travis-ci.org/github/python-pillow/docker-images/jobs/701583804#L3679))
> * Ubuntu 16.04 Xenial ([example](https://travis-ci.org/github/python-pillow/docker-images/jobs/701583801#L3264))

These are the only two test jobs running FreeType 2.6.x. It would be good to keep at least one of them to make sure FreeType tests have proper metrics for this version.

Amazon 1 and CentOS 6 run FreeType 2.3.1, and all other jobs run 2.8.x or newer.

File: .travis.yml [hunk 1]
```diff
@@ -36,15 +36,12 @@ matrix:
       services: xvfb
     - python: '3.7'
       name: "3.7 Xenial"
```
Comment (radarhere) [inline .travis.yml:38]:
```suggestion
      name: "3.7 Xenial PYTHONOPTIMIZE=2"
```

**Comment by jdufresne** [comment]
Using Python 3.6+ would allow the project to use f-strings if it is desired. IMO, f-strings tend to be nicer to read than `.format()` and `timeit` shows that they are slightly more performant.

`pyupgrade` can be used to adapt simple cases to use f-string. It only changes instances where the expression is shorter and does not introduce nested quotes. If you're interested, the command is

```
$ find . -name \*.py -exec pyupgrade --py36-plus {} \;
```

**Comment by hugovk** [comment]
Yep, was planning on doing that in a follow up. Wanted to get this smaller one in first to make it easier to review and reduce the CI sooner. Thanks!

File: docs/installation.rst [hunk 2]
```diff
@@ -19,7 +19,9 @@ Notes
 +--------------------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
 |**Python**          |**3.8**|**3.7**|**3.6**|**3.5**|**3.4**|**3.3**|**3.2**|**2.7**|**2.6**|**2.5**|**2.4**|
 +--------------------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
-|Pillow >= 7         |  Yes  |  Yes  |  Yes  |  Yes  |       |       |       |       |       |       |       |
+|Pillow >= 7.3       |  Yes  |  Yes  |  Yes  |       |       |       |       |       |       |       |       |
```
Comment (nulano) [inline docs/installation.rst:22]:
```suggestion
|Pillow >= 8.0       |  Yes  |  Yes  |  Yes  |       |       |       |       |       |       |       |       |
```

Also need to add Python 3.9 column, as it should be released the week before.

Comment (hugovk) [reply docs/installation.rst]: Thanks, done!

**Comment by hugovk** [comment]
@radarhere Is this okay to merge?

File: winbuild/build_prepare.py [hunk 3]
```diff
@@ -456,7 +456,7 @@ def build_pillow():
         cmd_cd("{pillow_dir}"),
         *prefs["header"],
         cmd_set("DISTUTILS_USE_SDK", "1"),  # use same compiler to build Pillow
-        cmd_set("MSSdk", "1"),  # for Python 3.5 and PyPy3.6
+        cmd_set("MSSdk", "1"),  # for PyPy3.6
         cmd_set("py_vcruntime_redist", "true"),  # use /MD, not /MT
```
Comment (nulano) [inline winbuild/build_prepare.py:460]: Just to note here so I don't forget:

`MSSdk` and `py_vcruntime_redist` will [no longer be required by PyPy](https://foss.heptapod.net/pypy/pypy/-/commit/1f5925b68f4210fbc5d852b03643c588d45f99a5) in the next release.

`py_vcruntime_redist` is still [required by CPython 3.6](https://github.com/python/cpython/blob/46cbf6148a46883110883488d3e9febbe46ba861/Lib/distutils/_msvccompiler.py#L268), but [not CPython 3.7](https://github.com/python/cpython/blob/c8c818b0d73680516d5841597b705a1feeb42113/Lib/distutils/_msvccompiler.py#L242).

**Comment by hugovk** [comment]
> > Not relevant for this particular PR, but two of the Docker images are using Python 3.5:
> > 
> > * Debian 9 Stretch ([example](https://travis-ci.org/github/python-pillow/docker-images/jobs/701583804#L3679))
> > * Ubuntu 16.04 Xenial ([example](https://travis-ci.org/github/python-pillow/docker-images/jobs/701583801#L3264))
> 
> ~These are the only two test jobs running FreeType 2.6.x.~ Xenial on Travis also has 2.6.1. It would be good to keep at least one of these after Xenial is EOL to make sure FreeType tests have proper metrics for this version.
> 
> Amazon 1 and CentOS 6 run FreeType 2.3.1, and all other jobs run 2.8.x or newer.

Let's drop these two for now, and can consider either upgrading one from Python 3.5 to something newer, or install FreeType 2.6.1 on another CI job.

More generally, we could pick a minimum version of FreeType to support. Some release dates [[1](https://www.freetype.org/)] [[2](https://lists.nongnu.org/archive/html/freetype-announce/2007-01/msg00002.html)]:

* 2020-05-09: 2.10.2 (newest)
* 2017-05-13: 2.8.0
* 2015-10-04: 2.6.1
* 2007-01-31: 2.3.1

**Pull request merged**